docs/DeveloperGuide.md

```diff
@@ -1,38 +1,536 @@
+
+![](./UML/Commands/Initial State.png)
+![](./UML/Commands/commandAddState.png)
+
+### List Command - Show all Questions with Index
+
+Lists all the questions in the current QuestionList. 
+
+When executed, the command will invoke the QuestionList.printQuestionList method. 
+This method first checks whether the list is empty and prints and error message if empty, 
+else it will invoke the QuestionList.printQuestion method on each Question object, 
+with the asList parameter set as true. The QuestionList.printQuestion method will then
+print each question with a given index, and indicate the question's type and completion status
+(obtained through Question.getQuestionType() and Question.questionIsDone() methods)
+
+### Delete Command - Delete a Question
+
+Deletes a question by its index.
+
+The Command first invokes the QuestionList.viewQuestionByIndex method to determine if 
+the index refers to a valid question, failing which it will return an error message.
+
+Otherwise, it will then invoke the QuestionList.deleteQuestionByIndex method to remove
+the question from the QuestionList. 
+
+![](./UML/Commands/commandDeleteQuestion.png)
+
+Afterward, the storage is updated to reflect the newest QuestionList.
+
+### Find Command - Look for a matching question
+
+Quizhub supports searching for specific questions, by searching matching keywords
+in the format
+
+`find /[description]` OR `find /[module]`
+i.e. `find /water buffalo`, `find /CS2113`
+
+This command is passed to the corresponding QuestionList where the `searchList` method
+is called. 
+This method go down the existing QuestionList of questions and use the Java contains method
+(along with the toLower method to ensure the search is case-insensitive)
+to determine if the search term is located that question. If the contains method returns true
+for a given question, that question's contents and index will be copied to a 
+new ArrayList of questions, and subsequently print them.
+
+### Edit Command - Edit Question / Answer
+#### Brief Description of Edit Command
+The CommandEdit class in the quizhub application is responsible for handling user commands to edit the description or 
+answer of a question. The CommandEdit class supports two edit commands: 
+
+- `edit [question number] /description [description]` - edits the description of the question with the specified number
+
+- `edit [question number] /answer [answer]` - edits the answer to the question with the specified number
+
+#### Class Structure of Edit Command
+The CommandEdit class includes the following key components:
+
+- `qnIndex`: An integer representing the question number to be edited.
+- `newDescription`: A string representing the new description for the question (or null if not edited).
+- `newAnswer`: A string representing the new answer for the question (or null if not edited).
+
+#### Implementation of Edit Command
+
+![commandEditSequence.png](UML%2FCommands%2FcommandEditSequence.png)
```

Comment:
In order to closely follow the convention learned in the lecture, you should delete the "calls" and "returns" statements. It could be visualised by using full vs. dotted lines instead
<img width="365" alt="image" src="https://github.com/nus-cs2113-AY2324S1/tp/assets/62311026/afaeb7ce-d3bd-412a-9b84-6bd100a09380">